subworkflows/local/prepare_references.nf

```diff
@@ -75,18 +78,42 @@ workflow PREPARE_REFERENCES {
         UNTAR_VEP_CACHE (ch_vep_cache)
         ch_versions = ch_versions.mix(UNTAR_VEP_CACHE.out.versions)
 
+        // Setting up Salmon index
+        ch_transcript_fasta = transcript_fasta ? Channel.fromPath( transcript_fasta ).collect() : Channel.empty()
+        if (transcript_fasta) {
+            if ( transcript_fasta.endsWith(".gz") ) {
+                ch_transcript_fasta_meta=Channel.fromPath(transcript_fasta).map{ it -> [ [id:it.simpleName], it ] }.collect()
+                ch_transcript_fasta_meta= GUNZIP_TRFASTA(ch_transcript_fasta_meta).gunzip
+                ch_transcript_fasta =  ch_transcript_fasta_meta.map{ meta, fasta -> [ fasta ] }
+                }
+        }
+
+        ch_salmon_index = salmon_index ? Channel.fromPath( salmon_index ).collect() : Channel.empty()
+        if ( !salmon_index ) {
+            if (!transcript_fasta) {
```

Comment:
```suggestion
        if ( !salmon_index && !transcript_fasta ) {
```

Reply:
It could be also possible to have the salmon_index but no transcript_fasta, so I think it should be 2 ifs